src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsprobationestateapi/service/GetProbationDeliveryUnitService.kt

```diff
@@ -0,0 +1,26 @@
+package uk.gov.justice.digital.hmpps.hmppsprobationestateapi.service
+
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+import uk.gov.justice.digital.hmpps.hmppsprobationestateapi.controller.dto.TeamOverview
+import uk.gov.justice.digital.hmpps.hmppsprobationestateapi.db.repositories.ProbationDeliveryUnitRepository
+import uk.gov.justice.digital.hmpps.hmppsprobationestateapi.db.repositories.TeamRepository
+import uk.gov.justice.digital.hmpps.hmppsprobationestateapi.exception.EntityNotFoundException
+
+@Service
+class GetProbationDeliveryUnitService(
+  private val probationDeliveryUnitRepository: ProbationDeliveryUnitRepository,
+  private val teamRepository: TeamRepository
+) {
+  fun findByCode(code: String): Flux<TeamOverview> =
```

Comment:
Not sure about referring to Flux in the service layer. For me it was make more sense to do the conversion to a flux (if indeed it is actually required) in the controller.

Reply:
Again, to answer my own question, this is part of the R2DBMS response so it makes sense to pass it thru